library/src/main/java/com/novoda/downloadmanager/lib/Constants.java

```diff
@@ -123,9 +123,14 @@
     public static final String MIMETYPE_APK = "application/vnd.android.package";
 
     /**
+     * The size of a tar block
+     */
+    public static final int BLOCK_SIZE = 512;
+
+    /**
      * The buffer size used to stream the data
      */
-    public static final int BUFFER_SIZE = 4096;
+    public static final int BUFFER_SIZE = 8 * BLOCK_SIZE;
```

Comment:
Ah I see. Nice! I usually go for two pages (8k) when doing disk I/O, but there's little difference I think.